File: 136.SingleNumber.py
from typing import List


class Solution:
    def singleNumber(self, nums: List[int]) -> int:
        single = nums[0]
        for num in nums[1:]:
            single ^= num
        return single


if __name__ == "__main__":
    s = Solution()
    nums = [4, 1, 2, 1, 2]  # 4
    print(s.singleNumber(nums))

File: bit/136.SingleNumber.go
package main

import "fmt"

func singleNumber(nums []int) int {
	num := nums[0]
	for _, value := range nums[1:] {
		num = num ^ value
	}
	return num
}

func main() {
	nums := []int{4, 1, 2, 1, 2}
	fmt.Println(singleNumber(nums))
}
